src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Microsoft.TestPlatform.VsTestConsole.TranslationLayer;
+using Microsoft.TestPlatform.VsTestConsole.TranslationLayer.Interfaces;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Serilog.Events;
+using Stryker.Core.ExecutableFinders;
+using Stryker.Core.Initialisation;
+using Stryker.Core.Options;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Stryker.Core.TestRunners.VsTest
+{
+    public partial class VsTestRunner : ITestRunner
+    {
+        private readonly StrykerOptions _options;
+        private readonly ProjectInfo _projectInfo;
+        private string _defaultRunSettings;
+        private int _timeoutMS = 0;
+        private readonly VsTestHelper _vsTestHelper;
+        private List<string> _messages = new List<string>();
+
+        public VsTestRunner(StrykerOptions options, ProjectInfo projectInfo)
+        {
+            _options = options;
+            _projectInfo = projectInfo;
+            _vsTestHelper = new VsTestHelper(options);
+
+            GenerateRunSettings();
+        }
+
+        public TestRunResult RunAll(int? timeoutMS, int? activeMutationId)
+        {
+            _timeoutMS = timeoutMS ?? 0;
+            GenerateRunSettings();
+
+            var testBinariesPath = FilePathUtils.ConvertPathSeparators(Path.Combine(_options.BasePath, "bin", "Debug", _projectInfo.TargetFramework));
+            var testAssemblyPath = FilePathUtils.ConvertPathSeparators(Path.Combine(testBinariesPath, _projectInfo.TestProjectFileName.Replace("csproj", "dll")));
+
+            var vsTestToolPath = _vsTestHelper.GetCurrentPlatformVsTestToolPath();
+            var vsTestExtensionsPath = _vsTestHelper.GetDefaultVsTestExtensionsPath(vsTestToolPath);
+
+            IVsTestConsoleWrapper consoleWrapper = null;
+
+            consoleWrapper = new VsTestConsoleWrapper(vsTestToolPath, new ConsoleParameters { TraceLevel = DetermineTraceLevel(), LogFilePath = Path.Combine(_options.BasePath, "StrykerOutput", "logs", "vstest-log.txt") });
+
+            consoleWrapper.StartSession();
+            consoleWrapper.InitializeExtensions(new List<string> { testBinariesPath, vsTestExtensionsPath });
+
+            var testCases = DiscoverTests(new List<string>() { testAssemblyPath }, consoleWrapper);
+
+            var testResults = RunAllTests(consoleWrapper, new List<string> { testAssemblyPath }, activeMutationId);
```

Comment:
I think all code above this line in this method can be moved to a other method. I think this needs to only be executed once, not for every mutant. This might speed up things too?